src/coreclr/gc/gc.cpp

```diff
@@ -43948,6 +43943,21 @@ HRESULT GCHeap::Initialize()
         return CLR_E_GC_LARGE_PAGE_MISSING_HARD_LIMIT;
     }
 
+#ifdef USE_REGIONS
+    gc_heap::regions_range = (size_t)GCConfig::GetGCRegionRange();
+    // TODO: Is there an easy way to check if the config is taking default value?
+    if (gc_heap::regions_range == 274877906944L)
+    {
+        if (gc_heap::heap_hard_limit)
+        {
+            gc_heap::regions_range = min(gc_heap::regions_range, 2 * gc_heap::heap_hard_limit);
+        }
+        gc_heap::regions_range = min(gc_heap::regions_range, 2 * gc_heap::total_physical_mem);
+        gc_heap::regions_range = align_on_page(gc_heap::regions_range);
+    }
```

Comment:
I am wondering about this. This code can adapt to various needs we will have. Including large physical machines and small containers. (of course, after changing the `regions_range` to be 0)
```suggestion
    if (gc_heap::regions_range == 0)
    {
        if (gc_heap::heap_hard_limit)
        {
            gc_heap::regions_range = 2 * gc_heap::heap_hard_limit;
        }
        else
        {
            gc_heap::regions_range = (size_t)(1.5 * gc_heap::total_physical_mem);
        }
        gc_heap::regions_range = align_on_page(gc_heap::regions_range);
    }
```